app/src/main/java/com/coderGtm/yantra/commands/theme/Command.kt

```diff
@@ -25,11 +27,68 @@ class Command(terminal: Terminal) : BaseCommand(terminal) {
             output(terminal.activity.getString(R.string.specify_theme), terminal.theme.errorTextColor)
             return
         }
-        if (args.size > 2) {
+        if (args.size > 3) {
             output(terminal.activity.getString(R.string.only_theme_name_reqd), terminal.theme.errorTextColor)
             return
         }
         val name = args[1].trim().lowercase()
+
+        if (name == "-s") {
+            if (args.size < 3) {
+                output("Please, specify a name of saving theme", terminal.theme.errorTextColor)
+                return
+            }
+
+            if (terminal.preferenceObject.getInt("theme", 0) != -1) {
+                output("Please, select a custom theme first", terminal.theme.errorTextColor)
+                return
+            }
+
+            val themeName = args[2].trim().lowercase()
+            val saved = terminal.preferenceObject.getString("theme$themeName",null)
+            if (Themes.entries.any { it.name.lowercase() == themeName } || themeName == "custom" || saved != null || themeName == "-s" || themeName == "-d") {
+                output("Theme already exists", terminal.theme.errorTextColor)
+                return
+            }
+
+            terminal.preferenceObject.edit().putString(
+                "theme$themeName", terminal.preferenceObject.getString(
+                "customThemeClrs",
+                "#FF000000,#FFA0A0A0,#FFE1BEE7,#FFFAEBD7,#FFEBEBEB,#FFF00000,#FF00C853,#FFFFD600"
```

Comment:
After commit #168 there are now 9 colors instead of 8. [Example](https://github.com/coderGtm/yantra-app-launcher/pull/168/files#diff-b94da87ca2811d521635ce3c217922409c8ddaa5ce6f9bca3aab5517fa46e860L345-R345)

Reply:
Ok so is this the only place to modify?

Reply:
yes